src/ui/pages/SidePage/components/WalletConnect/WalletConnect.tsx

```diff
@@ -1,16 +1,23 @@
-import { useState } from "react";
-import { useAppSelector } from "../../../../../store/hooks";
+import { useEffect, useState } from "react";
+import { useAppDispatch, useAppSelector } from "../../../../../store/hooks";
 import { getPendingConnection } from "../../../../../store/reducers/walletConnectionsCache";
 import { SideSlider } from "../../../../components/SideSlider";
 import { SidePageContentProps } from "../../SidePage.types";
 import { WalletConnectStageOne } from "./WalletConnectStageOne";
 import { WalletConnectStageTwo } from "./WalletConnectStageTwo";
+import { setPauseQueueIncomingRequest } from "../../../../../store/reducers/stateCache";
 
 const WalletConnect = ({ setOpenPage }: SidePageContentProps) => {
+  const dispatch = useAppDispatch();
   const pendingConnection = useAppSelector(getPendingConnection);
   const [requestStage, setRequestStage] = useState(0);
   const [hiddenStageOne, setHiddenStageOne] = useState(false);
 
+  useEffect(() => {
+    // Pause all incomming request when connect wallet opening
+    dispatch(setPauseQueueIncomingRequest(true));
```

Comment:
Hmm, imo we should only be calling this from the one place in the AppWrapper that is dependent on if we are logged in and online. Otherwise we will get weird timing bugs.

If the wallet is online, and logged in - and we pause here - what is unpausing it?

To cover the case where we have incoming requests from an old dApp appearing - when we pop it off the queue to display we should simply just not display anything and move on. Rather than pausing it

Reply:
It mean if user not connect to external wallet, we should dequeue all request from it right? And we don't need pause it.

Reply:
^ I was pretty sure there was already something in place from Patrick to cover that last case I mentioned but I need to dig into the code as I can't fully remember

Reply:
The user's connection to an external wallet should not impact the incoming request queue - it could be used for other things in the future.

So safest thing to do is to let the queue operate as normal based on the rules in the AppWrapper AND if we ever dequeue a cardano connect request that is from an old connection (does not match currently connected in Redux), we just move to the next item and don't display it

Reply:
`^ I was pretty sure there was already something in place from Patrick to cover that last case I mentioned but I need to dig into the code as I can't fully remember`
Yep. Patrick do it in `IncommingRequest` component

Reply:
Yeah looks good!

So as I understand the SidePage - it will:
- display an IncomingRequest if there is one
  - if the IncomingRequest relates to an old dApp connection, it skips it
- if there is no IncomingRequest but we are currently trying to connect to a dApp this will show

Possibly, it might make more sense to put the `WalletConnect` component as priority in case there is a lot of things in the queue? cc @sdisalvo-crd @jimcase

Reply:
Yep. Sure

Reply:
To me it makes sense, but lets make sure everyone thinks so too

Reply:
Done. Please help me check